mlir/lib/Dialect/LLVMIR/IR/LLVMDialect.cpp

```diff
@@ -2728,13 +2740,30 @@ LogicalResult LLVM::ConstantOp::verify() {
     if (isa<IntegerType>(getType()) && !getType().isInteger(floatWidth)) {
       return emitOpError() << "expected integer type of width " << floatWidth;
     }
-  }
-  if (auto splatAttr = dyn_cast<SplatElementsAttr>(getValue())) {
-    if (!isa<VectorType>(getType()) && !isa<LLVM::LLVMArrayType>(getType()) &&
-        !isa<LLVM::LLVMFixedVectorType>(getType()) &&
-        !isa<LLVM::LLVMScalableVectorType>(getType()))
+  } else if (isa<ElementsAttr, ArrayAttr>(getValue())) {
+    if (isa<LLVM::LLVMScalableVectorType>(getType())) {
```

Comment:
I was also wondering about that. We have test cases such as:
```
mlir/test/Conversion/MathToLLVM/math-to-llvm.mlir:  // CHECK: %[[ONE:.*]] = llvm.mlir.constant(dense<1.000000e+00> : vector<[4]xf32>) : vector<[4]xf32>
mlir/test/Dialect/ArmSVE/legalize-for-llvm.mlir:  // CHECK-NEXT:    %[[RES0:.*]] = llvm.mlir.constant(dense<false> : vector<2x[16]xi1>) : !llvm.array<2 x vector<[16]xi1>>
```

This is what the op documentation says: `The operation produces a new SSA value of the specified LLVM IR dialect type. The type of that value _must_ correspond to the attribute type converted to LLVM IR.
`

Reply:
I guess that this is just UB when rhe vector size does not match. I usualy check what LLVM's verifier catch for such cases, but scalable vector support might not be too great.

Reply:
The arith constant seems to check if the attribute is a splat elements attribute. A size check is not needed in the arith case since result and value type are equivalent... 

I would suggest to keep the PR as is for now except you are up to figure out the exact semantics here. I assume the right thing would be to check for a splat elements attribute and matching scalable sizes?

Reply:
@banach-space @dcaballe Do you know how we should verify the number of elements when we have vector types and/or attributes with scalable dimensions?

Reply:
I've looked at this in a slightly different context recently. My recommendation is to only allow "splats" when dealing with scalable vectors. As in, this would be fine:
```mlir
llvm.mlir.constant(dense<[0]> : vector<[4]xi32>)
```
but not this:
```mlir
llvm.mlir.constant(dense<[0, 1, 2, 3]> : vector<[4]xi32>)
```

I don't envisage us requiring the 2nd option.

Reply:
I added another check to the verifier such that a scalable vector type requires a splat attribute.

Reply:
Nice!